### PR TITLE
[WIP] Fix linebreaking

### PR DESCRIPTION
**Pull request by stefan-kolb**
Evaluating the usage and problems of #114.
What is FieldContentParser really doing? What is JabRef style formatting?

### CODE REVIEW

File: src/main/java/net/sf/jabref/importer/fileformat/FieldContentParser.java [hunk 1]
```diff
@@ -18,28 +18,44 @@
...
+     *
+     * @param content StringBuffer containing the field to format. bibtexKey contains field name according to field
+     *                was edited by Kuehn/Havalevich
+     * @param bibtexKey
+     * @return The formatted field content. The StringBuffer returned may or may not be the same as the argument given.
      */
-    public StringBuffer format(StringBuffer content, String key) {
-        int i = 0;
-
+    public StringBuffer format(StringBuffer content, String bibtexKey) {
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/importer/fileformat/FieldContentParser.java]: I think the name `bibtexKey` is misleading, as I would think of it as the key of a bibtex entry. I would suggest to use `bibtexField` instead.

**Comment by lenhard** [comment]
The remaining code of `FileContentParser.format` is a complete mess of one-off hacks. I am very much for deleting it and rewriting the parsing logic to just remove newlines for non-multiline fields and ignore tabs or spaces completely. If we document that with tests, at least we know what the code is doing...

**Comment by lenhard** [comment]
859274e54af0948108ff16fa697dabb595768c12 replaces the weird code in `FieldContentParser` with a single (and simple) regex. Now the tests pass and, as far as I can tell, #114 and #186 are fixed. Could someone else please double check? (Note: Parsing is only done when loading the file. So if you add faulty text to a field, you can save it as such. On restart, the errors should be fixed by the parser). 

If something is not quite right, please add the faulty behavior as a test!

File: src/test/java/net/sf/jabref/importer/fileformat/BibtexParserTest.java [hunk 2]
```diff
@@ -309,7 +309,7 @@ public void testFileNaming() {
                 + "file = {Bemerkung:H:\\bla\\ups  sala.pdf:PDF}, \n"
                 + "}");
 
-        Assert.assertEquals("Bemerkung:H:\\bla\\ups  sala.pdf:PDF", e.getField("file"));
+        Assert.assertEquals("Bemerkung:H:\\bla\\ups sala.pdf:PDF", e.getField("file"));
```
Comment (stefan-kolb) [inline src/test/java/net/sf/jabref/importer/fileformat/BibtexParserTest.java]: You break functionality here, when a file path contains multiple whitespaces.

**Comment by lenhard** [comment]
thanks @stefan-kolb! the "file" field is now ignored from parsing.

**Comment by stefan-kolb** [comment]
Ok I think we are goog to go. You can also close all other mentioned issues for now!

**Comment by koppor** [comment]
Works great here :+1:

**Pull request merged**